dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java

```diff
@@ -1,50 +1,80 @@
+import com.dotcms.business.SystemTableUpdatedKeyEvent;
 import com.dotcms.filters.interceptor.Result;
 import com.dotcms.filters.interceptor.WebInterceptor;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.system.event.local.model.EventSubscriber;
 import com.dotcms.util.CollectionsUtils;
 import com.dotcms.util.WhiteBlackList;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.UUIDUtil;
 import com.liferay.util.StringPool;
+import io.vavr.control.Try;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Predicate;
 
 /**
  * Web Interceptor to track analytics
  * @author jsanca
  */
-public class AnalyticsTrackWebInterceptor  implements WebInterceptor {
+public class AnalyticsTrackWebInterceptor  implements WebInterceptor, EventSubscriber<SystemTableUpdatedKeyEvent> {
 
+    private final static String ANALYTICS_TURNED_ON_KEY = "ANALYTICS_TURNED_ON";
     private final static Map<String, RequestMatcher> requestMatchersMap = new ConcurrentHashMap<>();
+    private final HostWebAPI hostWebAPI;
+    private final AppsAPI appsAPI;
 
     /// private static final String[] DEFAULT_BLACKLISTED_PROPS = new String[]{"^/api/*"};
     private static final String[] DEFAULT_BLACKLISTED_PROPS = new String[]{StringPool.BLANK};
-    private final WhiteBlackList whiteBlackList = new WhiteBlackList.Builder()
-            .addWhitePatterns(Config.getStringArrayProperty("ANALYTICS_WHITELISTED_KEYS",
-                    new String[]{StringPool.BLANK})) // allows everything
-            .addBlackPatterns(CollectionsUtils.concat(Config.getStringArrayProperty(  // except this
-                    "ANALYTICS_BLACKLISTED_KEYS", new String[]{}), DEFAULT_BLACKLISTED_PROPS)).build();
+    private final WhiteBlackList whiteBlackList;
+    private final AtomicBoolean isTurnedOn;
 
     public AnalyticsTrackWebInterceptor() {
 
-        addRequestMatcher(
+        this(WebAPILocator.getHostWebAPI(), APILocator.getAppsAPI(),
+                new WhiteBlackList.Builder()
+                        .addWhitePatterns(Config.getStringArrayProperty("ANALYTICS_WHITELISTED_KEYS",
+                                new String[]{StringPool.BLANK})) // allows everything
+                        .addBlackPatterns(CollectionsUtils.concat(Config.getStringArrayProperty(  // except this
+                                "ANALYTICS_BLACKLISTED_KEYS", new String[]{}), DEFAULT_BLACKLISTED_PROPS)).build(),
+                new AtomicBoolean(Config.getBooleanProperty(ANALYTICS_TURNED_ON_KEY, true)),
```

Comment:
Let's rename it to our Feature Flag convention: 
`FEATURE_FLAG_CONTENT_ANALYTICS`